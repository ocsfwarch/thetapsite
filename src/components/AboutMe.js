import React from "react";

const AboutMe = () => {
  return (
    <div className="aboutme_container">
      <section className="img_sect">
        <section className="img_display">
          <img
            src="/images/AboutMe/tp_image_2.png"
            title="About Me"
            alt="About Me"
          />
        </section>
      </section>
      <section className="intro_sect">
        <p>
          My name is Terry Partridge. I have been a Full Stack Software Engineer
          based in Mission Viejo, California since 2000. For the past 2 years my
          primary focus for frontend and backend development has been the
          PostgreSQL, Express, React, and Node (PERN) stack. I love the creative
          process associated with developing software and the constant advances
          in software development technology. It allows me to build applications
          that help people and it feeds my need for continuous improvement.
        </p>
      </section>
      <section className="footer_sect">
        <section className="app_header_menu">
          <nav className="navContent">
            <section>
              <ul className="navLinks">
                <li>Work History</li>
                <li>Skills</li>
                <li>Education</li>
              </ul>
            </section>
          </nav>
        </section>
      </section>
    </div>
  );
};

export { AboutMe };
