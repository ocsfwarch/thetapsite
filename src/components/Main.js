import React from "react";
import pic from "../images/IMG_9450_small.png";

export const Main = () => {
  return (
    <div className="intro_container">
      <section className="intro_img_container">
        <img src={pic} alt="display" />
        <span className="caption">Grand Canyon AZ from Plateau Point</span>
      </section>
      <section className="intro_content_container">
        <section className="banner">
          <h1>Welcome to my World Wide Web space</h1>
        </section>
        <section className="content">
          <p>
            My name is Terry Partridge. I am a Software Engineer based in
            Mission Viejo, California. This is my space, I hope you find it
            useful.
          </p>
        </section>
      </section>
    </div>
  );
};
