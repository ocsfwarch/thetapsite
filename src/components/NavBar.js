import React from "react";

export const NavBar = () => {
  return (
    <nav className="navContent">
      <section>
        <ul className="navLinks">
          <li>
            <a href="/">Home</a>
          </li>
          <li>
            <a href="/Projects">Projects</a>
          </li>
          <li>
            <a href="/AboutMe">About Me</a>
          </li>
        </ul>
      </section>
    </nav>
  );
};
