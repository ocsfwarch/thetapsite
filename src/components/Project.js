import React from 'react';

export const Project = ({project}) => {
    //console.log(`path=${window.location.pathname}, Image = ${img}`)
    const openProject = () => {
        window.open(project.href, '_blank');
    }

    const techStack = project.stack.map((item,idx) => {
        return(<img key={idx} src={item.src}        title={item.title}      alt={item.alt}></img>)
    });

    return(
        <div className="project_container">
            <section className="project_title">
                <a href={project.href} target="_blank" title="Click to show the app">{project.title}</a>
            </section>
            
            <section className="project_image">
                <section className="project_image_container">
                    <img src={project.image} />
                </section>
            </section>
            <section className="project_summary">
                <p>
                    {project.summary}
                </p>
                <p>
                    {techStack}
                </p>
            </section>

            <section className="project_footer">
                <button type="button" className="btn btn-primary" onClick={openProject}>Visit App</button>
            </section>
        </div>
    );
};
