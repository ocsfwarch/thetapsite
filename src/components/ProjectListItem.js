import React from "react";

export const ProjectListItem = ({ project }) => {
  const techStack = project.stack.map((item, idx) => {
    return (
      <img key={idx} src={item.src} title={item.title} alt={item.alt}></img>
    );
  });

  let appLink = "";
  if (project.href && project.href.length > 0) {
    appLink = (
      <a
        href={project.href}
        target="_blank"
        rel="noreferrer"
        title="Click to show the app"
      >
        <button type="button" className="btn btn-primary">
          Visit App
        </button>
      </a>
    );
  }

  let gitLink = "";
  if (project.git && project.git.length > 0) {
    gitLink = (
      <a
        href={project.git}
        target="_blank"
        rel="noreferrer"
        title="Click to visit the GitHub repository"
      >
        <button type="button" className="btn btn-primary">
          Visit GitHub
        </button>
      </a>
    );
  }

  return (
    <div className="project_container">
      <section className="project_title">
        <a
          href={project.href}
          target="_blank"
          rel="noreferrer"
          title="Click to show the app"
        >
          {project.title}
        </a>
      </section>

      <section className="project_image">
        <img src={project.image} alt="" />
      </section>

      <section className="project_summary">
        <p>{project.summary}</p>
      </section>

      <section className="project_stack">{techStack}</section>

      <section className="project_footer">
        {appLink}
        {gitLink}
      </section>
    </div>
  );
};
