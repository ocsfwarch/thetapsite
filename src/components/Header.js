import React from 'react';
import { NavBar } from './NavBar';

export const Header = () => {

    return(
        <div className="app_header">
            <div className="app_header_banner">
                <section ><h1>The TAP Site</h1></section>
            </div>
            <div className={"app_header_menu"}>
                <NavBar />
            </div>
        </div>
    );
};