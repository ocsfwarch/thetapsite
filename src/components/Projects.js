import React from "react";
import { Project } from "./Project";
import { ProjectItems } from "./Helpers/ProjectItems";

export const Projects = () => {
  const projectList = ProjectItems.map((item, idx) => (
    <Project key={idx} project={item} />
  ));
  return <div className="projects_container">{projectList}</div>;
}; // End Projects
