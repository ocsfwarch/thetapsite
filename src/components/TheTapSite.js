import React from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from "react-router-dom";
import { Header } from "./Header";
import { Main } from "./Main";
import { Projects } from "./Projects";
import { AboutMe } from "./AboutMe/AboutMe";
import { WorkHistory } from "./AboutMe/WorkHistory";
import { Skills } from "./AboutMe/Skills";
import { Education } from "./AboutMe/Education";

function TheTapSite() {
  return (
    <Router>
      <div className="app_container">
        <div className="app_content_container">
          <Header />
          <Switch>
            <Route
              exact
              path={`${process.env.PUBLIC_URL}/`}
              render={() => (
                <React.Fragment>
                  <Main />
                </React.Fragment>
              )}
            />
            <Route
              exact
              path={`${process.env.PUBLIC_URL}/Projects`}
              component={Projects}
            />
            <Route
              exact
              path={`${process.env.PUBLIC_URL}/AboutMe`}
              component={AboutMe}
            />
            <Route
              path={`${process.env.PUBLIC_URL}/WorkHistory`}
              component={WorkHistory}
            />
            <Route
              path={`${process.env.PUBLIC_URL}/Skills`}
              component={Skills}
            />
            <Route
              path={`${process.env.PUBLIC_URL}/Education`}
              component={Education}
            />
            <Redirect to={`${process.env.PUBLIC_URL}/`} />
          </Switch>
        </div>
      </div>
    </Router>
  );
}

export default TheTapSite;
