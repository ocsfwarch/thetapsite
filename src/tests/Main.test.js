import React from "react";
import renderer from "react-test-renderer";
import { Main } from "../components/Main";

it("renders correctly", () => {
  const tree = renderer.create(<Main />).toJSON();
  expect(tree).toMatchSnapshot();
});
