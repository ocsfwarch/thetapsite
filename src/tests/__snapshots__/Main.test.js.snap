// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="intro_container"
>
  <section
    className="display_img"
  >
    <section
      className="display_img_image"
    >
      <img
        src="IMG_9450_small.png"
      />
    </section>
    <section
      className="display_img_caption"
    >
      <span
        className="caption"
      >
        Grand Canyon AZ from Plateau Point
      </span>
    </section>
  </section>
  <section
    className="intro_content"
  >
    <section
      className="banner"
    >
      <h1>
        Welcome to my World Wide Web space
      </h1>
    </section>
    <article
      className="content"
    >
      <p>
        My name is Terry Partridge. I am a Software Engineer based in Mission Viejo, California. This is my space, I hope you find it useful.
      </p>
    </article>
  </section>
</div>
`;
